cmake/CppInterOp/CppInterOpConfig.cmake.in

```diff
@@ -44,7 +46,7 @@ set(CPPINTEROP_LLVM_VERSION_MAJOR "@LLVM_VERSION_MAJOR@")
 set(CPPINTEROP_LLVM_VERSION_MINOR "@LLVM_VERSION_MINOR@")
 set(CPPINTEROP_LLVM_VERSION_PATCH "@LLVM_VERSION_PATCH@")
 set(CPPINTEROP_LLVM_VERSION_SUFFIX "@LLVM_VERSION_SUFFIX@")
-set(CPPINTEROP_LLVM_PACKAGE_VERSION "@PACKAGE_VERSION@")
+set(CPPINTEROP_LLVM_PACKAGE_VERSION "@CPPINTEROP_VERSION@")
```

Comment:
Hmm I mean the names are a bit confusing and the functionality also depicts otherwise :|

So I see this 
https://github.com/compiler-research/CppInterOp/blob/6ab4d34b9633a0688a6582efd01cdf83ba78991f/cmake/CppInterOp/CppInterOpConfig.cmake.in#L42

Which says `CPPINTEROP_LLVM_VERSION` gives me the LLVM version CppInterOp is built on.

And getting cppinterop from conda-forge I can see the following in `CppInterOpConfigVersion.cmake`
```
set(PACKAGE_VERSION "1.4.0")

if("${PACKAGE_FIND_VERSION_MAJOR}.${PACKAGE_FIND_VERSION_MINOR}" VERSION_LESS "1.4"
   AND NOT "0" VERSION_LESS "${PACKAGE_FIND_VERSION_PATCH}")
  set(PACKAGE_VERSION_COMPATIBLE 1)
endif()
if("1.4.0" VERSION_EQUAL "${PACKAGE_FIND_VERSION_MAJOR}.${PACKAGE_FIND_VERSION_MINOR}.${PACKAGE_FIND_VERSION_PATCH}")
    set(PACKAGE_VERSION_EXACT 1)
endif()
```
Which says `PACKAGE_VERSION` maps to `1.4.0`

Now in `CppInterOpConfig.cmake.in` I don't think `PACKAGE_VERSION` is accessible so I changed it with `CPPINTEROP_VERSION` which also maps to `1.4.0`

Reply:
I mean we have that
That isn't an issue 

`CPPINTEROP_LLVM_VERSION` gives us this.

This is what we have finally if you let's say get cppinterop from an source like conda-forge/emscripten-forge etc
```
# This file allows users to call find_package(CppInterOp) and pick up our targets.

# Compute the installation prefix from this CppInterOpConfig.cmake file location.
get_filename_component(CPPINTEROP_INSTALL_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
get_filename_component(CPPINTEROP_INSTALL_PREFIX "${CPPINTEROP_INSTALL_PREFIX}" PATH)
get_filename_component(CPPINTEROP_INSTALL_PREFIX "${CPPINTEROP_INSTALL_PREFIX}" PATH)
get_filename_component(CPPINTEROP_INSTALL_PREFIX "${CPPINTEROP_INSTALL_PREFIX}" PATH)

# Set CPPINTEROP_CONFIG, CPPINTEROP_DIR, and CPPINTEROP_VERSION
set(CPPINTEROP_CONFIG "${CPPINTEROP_INSTALL_PREFIX}/lib/cmake/cppinterop")
set(CPPINTEROP_DIR "${CPPINTEROP_INSTALL_PREFIX}")
set(CPPINTEROP_VERSION "1.4.0")  # Make sure this matches the version in CppInterOpConfigVersion.cmake

# Determine CMAKE_SHARED_LIBRARY_SUFFIX based on operating system
include(CMakeSystemSpecificInformation)

if(MSVC)
  set(shared_lib_dir bin)
else()
  set(shared_lib_dir lib)
endif()

### build/install workaround
if (ON)
  set(_lib_suffix ${CMAKE_SHARED_LIBRARY_SUFFIX})
  set(_lib_prefix ${CMAKE_SHARED_LIBRARY_PREFIX})
else()
  set(_lib_suffix ${CMAKE_STATIC_LIBRARY_SUFFIX})
  set(_lib_prefix ${CMAKE_STATIC_LIBRARY_PREFIX})
endif()

if (IS_DIRECTORY "${CPPINTEROP_INSTALL_PREFIX}/include")
  set(_include "${CPPINTEROP_INSTALL_PREFIX}/include")
  set(_lib "${CPPINTEROP_INSTALL_PREFIX}/${shared_lib_dir}/${_lib_prefix}clangCppInterOp${_lib_suffix}")
else()
  set(_include "/Users/runner/miniforge3/conda-bld/cppinterop_1729512951783/work/cppinterop/include")
  set(_lib "/Users/runner/miniforge3/conda-bld/cppinterop_1729512951783/work/cppinterop/build/${shared_lib_dir}/${_lib_prefix}clangCppInterOp${_lib_suffix}")
endif()

###

set(CPPINTEROP_EXPORTED_TARGETS "clangCppInterOp")
set(CPPINTEROP_CMAKE_DIR "${_cmake}")
set(CPPINTEROP_INCLUDE_DIRS "${_include}")
set(CPPINTEROP_LIBRARIES "${_lib}")

set(CPPINTEROP_LLVM_VERSION "19.1.2")
set(CPPINTEROP_LLVM_VERSION_MAJOR "19")
set(CPPINTEROP_LLVM_VERSION_MINOR "1")
set(CPPINTEROP_LLVM_VERSION_PATCH "2")
set(CPPINTEROP_LLVM_VERSION_SUFFIX "")
set(CPPINTEROP_LLVM_PACKAGE_VERSION "")

# Provide all our library targets to users.
if (ON)
  add_library(clangCppInterOp SHARED IMPORTED)
else()
  add_library(clangCppInterOp STATIC IMPORTED)
endif()
set_target_properties(clangCppInterOp PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES ${_include}
  IMPORTED_LOCATION ${_lib}
  )
if (MSVC)
  if (IS_DIRECTORY "${CPPINTEROP_INSTALL_PREFIX}/include")
    set(_static_lib "${CPPINTEROP_INSTALL_PREFIX}/lib/${_lib_prefix}clangCppInterOp.lib")
  else()
    set(_static_lib "/Users/runner/miniforge3/conda-bld/cppinterop_1729512951783/work/cppinterop/build/lib/${_lib_prefix}clangCppInterOp.lib")
  endif()

  set_target_properties(clangCppInterOp PROPERTIES
    IMPORTED_IMPLIB ${_static_lib}
    )
endif(MSVC)

# Echo the variables to check their values
message(STATUS "CPPINTEROP_CONFIG: ${CPPINTEROP_CONFIG}")
message(STATUS "CPPINTEROP_CMAKE_DIR: ${CPPINTEROP_CMAKE_DIR}")
message(STATUS "CPPINTEROP_LIBRARIES: ${CPPINTEROP_LIBRARIES}")
message(STATUS "CPPINTEROP_DIR: ${CPPINTEROP_DIR}")
message(STATUS "CPPINTEROP_VERSION: ${CPPINTEROP_VERSION}")

unset(_lib_prefix)
unset(_lib_suffix)
unset(_cmake)
unset(_include)
unset(_lib)
```

Reply:
Check the variables set 
1) the `_cmake` one would end up being emtpy which i fix above
2) `CPPINTEROP_VERSION` is set to `1.4.0` which i use above

Reply:
Then I have no clue what's the intent of `CPPINTEROP_LLVM_PACKAGE_VERSION`. Maybe @alexander-penev remembers...

Reply:
So in this case, we know how we can proceed @vgvassilev 

1) > CPPINTEROP_LLVM_PACKAGE_VERSION should be the version of LLVM with which CppInterOp was compiled

So basically we have the following.
```
set(CPPINTEROP_LLVM_VERSION "@LLVM_VERSION@")
....
+ set(CPPINTEROP_LLVM_PACKAGE_VERSION "@CPPINTEROP_LLVM_VERSION@")
```
There is duplication here but if we want `CPPINTEROP_LLVM_PACKAGE_VERSION` to depict the llvm version, this is how it would be done.

2) We expose cppinterop's version not through `CPPINTEROP_LLVM_PACKAGE_VERSION` but through `CPPINTEROP_VERSION` itself.

Reply:
@alexander-penev I've applied the changes as per your reply. let me know if this is what you were expecting !